mtp_cashbook/apps/cashbook/tests/test_forms.py

```diff
@@ -0,0 +1,122 @@
+import mock
+
+from django.test.testcases import SimpleTestCase
+
+from cashbook.forms import ProcessTransactionBatchForm
+
+
+class ProcessTransactionBatchFormTestCase(SimpleTestCase):
+
+    def setUp(self):
+        super(ProcessTransactionBatchFormTestCase, self).setUp()
+        self.user = mock.MagicMock()
+        self.request = mock.MagicMock()
+
+        self.pending_transactions_response = {
+            'count': 4,
+            'results': [{'id': i} for i in range(1, 5)]
+        }
+
+    @mock.patch('cashbook.forms.get_connection')
+    def __call__(self, result, mocked_get_connection, *args, **kwargs):
+        self.mocked_get_connection = mocked_get_connection
+        super(ProcessTransactionBatchFormTestCase, self).__call__(result, *args, **kwargs)
+
+    def test_init_form_with_already_locked_transactions(self):
+        self.mocked_get_connection().\
+            transactions()().\
+            get.return_value = self.pending_transactions_response
+
+        form = ProcessTransactionBatchForm(self.request)
+
+        expected_choices = [
+            (t['id'], t) for t in self.pending_transactions_response['results']
+        ]
+        self.assertEqual(form.transaction_choices, expected_choices)
+        self.assertEqual(form.fields['transactions'].choices, expected_choices)
+
+    def test_init_form_with_no_locked_transactions(self):
+        self.mocked_get_connection().transactions()().get.side_effect = [
+            {
+                'count': 0,
+                'results': 0
+            },
+        ] + [self.pending_transactions_response]
+
+        form = ProcessTransactionBatchForm(self.request)
+        expected_choices = [
+            (t['id'], t) for t in self.pending_transactions_response['results']
+        ]
+        self.assertEqual(form.transaction_choices, expected_choices)
+        self.assertEqual(form.fields['transactions'].choices, expected_choices)
+
+    def test_credit_some(self):
```

Comment:
We test crediting some, and discarding all; is it worth testing crediting all, too?